src/sql/src/plan.rs

```diff
@@ -564,23 +564,27 @@ impl QueryWhen {
         }
     }
     /// Returns whether the candidate must be advanced to the since.
-    pub fn advance_to_since(&self) -> bool {
+    pub fn advance_to_since(&self, strict_serializable: bool) -> bool {
         match self {
-            QueryWhen::Immediately | QueryWhen::AtLeastTimestamp(_) => true,
+            QueryWhen::Immediately | QueryWhen::AtLeastTimestamp(_) => !strict_serializable,
```

Comment:
What's the importance of this returning false when in SS?

Reply:
I'm unsure about `AtLeastTimestamp`. On the one hand, the user has specified `AS OF` which removes them from SS land entirely, so seems like this should still return true. On the other hand, `advance_to_global_ts` will still bump it forward to a time that should always be at least the since? I think I'm more convinced by the argument of "`AS OF` never interacts with serializability requirements" and have the code also reflect that (never have `strict_serializable` appear after `AtLeastTimestamp`. But! I think `advance_to_global_ts` could choose to return true for `AtLeastTimestamp` no matter what? Hmm.

Reply:
> What's the importance of this returning false when in SS?

My thinking here is that the global read timestamp should always be larger than or equal to `since` due to our read holds. However if for whatever reason this isn't true, I'd rather things blow up than silently violate consistency guarantees.

Reply:
I'm still thinking about your second comment, I haven't ignored it. Though I may not get to it today.

Reply:
So I cleaned up these methods and removed all parameters. I think isolation level and table usage logic should be kept in the Coord and not in the plan module. Now these methods are pretty simple and identical though. Let me know what you think of the change.